### PR TITLE
sql: sequences: remove CYCLE option

### PR DESCRIPTION
**Pull request by vilterp**
Makes using `CYCLE` return an `unimplemented` error.

It requires changes to the KV layer that just don't seem worth the time right now, since it's a strange behavior and I suspect most users want the default behavior (`NO CYCLE`) anyway. (Discussed with @vivekmenezes).

FYI @lhirata; sorry to pull this out once you had already documented it.

Release note: None

### CODE REVIEW

**Comment by spencerkimball** [comment]
@vilterp I think we should work to support this – I really hate to see us deviate from Postgres as this will undoubtedly break things in difficult-to-workaround ways. It'd be one thing if this wasn't simple to support, but it requires pretty minimal changes (I think).

I'll send you suggested changes to the KV layer to support MIN / MAX / CYCLE options on the `Increment` command. Nothing in the system uses increment, so expanding its purview carries little risk. Might as well tailor it to suit sequences.

**Comment by vivekmenezes** [comment]
@vilterp any chance this can be implemented at the SQL layer using 

```
val, fail := incr()
if fail {
 cput(minValue, maxValue)
return minValue
}
return val
```

**Comment by bdarnell** [comment]
I understand the use of sequences to construct unique IDs. I don't understand the use case for cycling sequences that guarantee neither uniqueness nor monotonicity. Do we have evidence that this is widely used? It seems to me that this is far down the list of our postgres incompatibilities, and we'd be better served by having simpler non-cycling sequences and moving on than polishing this feature to perfect compatibility.

**Comment by vilterp** [comment]
@spencerkimball I got KV-layer changes working and mostly reviewed in this PR (https://github.com/cockroachdb/cockroach/pull/20574). May get back to it soon. Just was getting nervous about making sure that the other, more important items on [the list](https://github.com/cockroachdb/cockroach/projects/13) (load/dump, prevent dropping if used) get in for 2.0, and being available to help with admin UI. One complication: mixed-version clusters. Have to add another version, and a check to prevent sequence creation if not all nodes support the new `Inc` behavior.

@vivekmenezes have to do the read & the write in the same txn, which requires more KV roundtrips.

**Comment by spencerkimball** [comment]
@vilterp fair enough. Keep in mind that the mixed-version cluster problem is not really a problem since when folks upgrade to 2.0, they're going to have to change the cluster version setting to get most of the new functionality, and sequences with cycles can definitely be part of that upgraded functionality. I guess I'm just saying that part of the problem isn't going to get any easier next cycle.

File: pkg/sql/sqlbase/structured.proto [hunk 1]
```diff
@@ -621,6 +621,7 @@ message TableDescriptor {
     // Start value of the sequence.
     optional int64 start = 4 [(gogoproto.nullable) = false];
     // Whether to wrap around when the min or max value is hit.
+    // Currently unimplemented (doesn't parse, value ignored).
```
Comment (justinj) [inline pkg/sql/sqlbase/structured.proto]: Any reason to keep this field around? If you decide to keep it around I'd mark yourself with a `TODO(vilterp)` so whoever comes across it knows who to ask for context.

**Review by justinj — APPROVED** [review]
LGTM

**Pull request merged**